### PR TITLE
ci(releases): constrain PyPI GitHub Actions release trigger types

### LINKED ISSUE #456: Avoid errors with multiple publish jobs on publishing GitHub releases
I noticed there were several jobs triggered from the publishing of a GitHub Release lately. Opening this issue to see about avoiding these occurrences if possible.

Succeeded:
- https://github.com/cytomining/pycytominer/actions/runs/11115084541

Failed:
- https://github.com/cytomining/pycytominer/actions/runs/11115084548
- https://github.com/cytomining/pycytominer/actions/runs/11115084555

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR seeks to fix the GitHub Actions errors which followed a recent release. There are many different trigger types for releases, I believe we saw the job triggered from several of the types (see here for more: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release ). 

Closes #456 

## What is the nature of your change?

- [x] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: .github/workflows/pypi-release.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 name: Publish Python 🐍 distributions 📦 to PyPI and TestPyPI
-on: [release]
+on:
+  release:
+    # only publish to pypi when gh release is published
+    # (there are many gh release triggers).
```
Comment (gwaybio) [inline .github/workflows/pypi-release.yml:5]: I'm not sure what you mean by this? If there are many, do they do things differently? Why is it important to note that there are many?

**Review by gwaybio — APPROVED** [review]
One comment to potentially clarify

File: .github/workflows/pypi-release.yml (hunk 1 above)
Comment (d33bs) [reply .github/workflows/pypi-release.yml:5]: Thanks - I added this as a reference to this moment where we saw the job triggered from many types of release triggers. I'll remove the mention here as I think the meaning might be lost in time and become confusing.

**Comment by d33bs** [comment]
Thanks @gwaybio for the review! I made a small modification based on your comment. Merging this in now.

**Pull request merged**